### PR TITLE
feat(expression router) move setting protocols back to kongstate override for expression based routes

### LINKED ISSUE #4313: Fill `protocol` field in Kong routes when translating to expression routes
As suggestion from gateway team, it is better to put protocols in `protocols` field of Kong routes, instead of adding `net.protocol == ***` in expressions. Currently the parser will put protocol matchers in expressions, including adding the default `(net.protocol == "http") || (net.protocol) == “https”`. We should move the matching rules of protocols into `protocols` field of translated Kong routes, as we did for traditional router.

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

As suggested by gateway team, it is better to configure the protocols of Kong routes in `protocols` field than filling them directly into the expression. This PR moves the filling protocol back to the `kongstate` override phase of generated Kong routes, as with traditional router.  

**Which issue this PR fixes**:

fixes #4313 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/parser/translate_httproute_test.go
Comment (pmalek) [inline internal/dataplane/parser/translate_httproute_test.go:1]: Should we add a test when `protocol` is used in the expression? I gather that this should return an error right? Why not test it?

File: internal/dataplane/parser/translators/ingress_atc_test.go
Comment (pmalek) [inline internal/dataplane/parser/translators/ingress_atc_test.go:1]: Ditto here: can we add a test where `protocol` is used?

**Comment by czeslavo** [comment]
Allowed myself to rebase onto main and regenerate golden files (needed after https://github.com/Kong/kubernetes-ingress-controller/pull/4426).

File: internal/dataplane/parser/translators/httproute_atc.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func GenerateKongExpressionRoutesFromHTTPRouteMatches(
 
 	// if we do not need to generate a kong route for each match, we OR matchers from all matches together.
 	routeMatcher := atc.And(atc.Or(generateMatchersFromHTTPRouteMatches(translation.Matches)...))
-	// add matcher from parent httproute (hostnames, protocols, SNIs) to be ANDed with the matcher from match.
+	// add matcher from parent httproute (hostnames, SNIs) to be ANDed with the matcher from match.
```
Comment (programmer04) [inline internal/dataplane/parser/translators/httproute_atc.go:62]:
```suggestion
	// Add matcher from parent httproute (hostnames, SNIs) to be ANDed with the matcher from match.
```

Comment (randmonkey) [reply internal/dataplane/parser/translators/httproute_atc.go:62]: done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -157,6 +157,9 @@ Adding a new version? You'll need three changes:
   to `/status/ready`. Gateways will be considered ready only after an initial
   configuration is applied by the controller.
   [#4368](https://github.com/Kong/kubernetes-ingress-controller/pull/4368
+- Moved protocol overriding to `kongstate.Override` step for expression based
+  routes, instead of adding expressions to match protocols.
```
Comment (pmalek) [inline CHANGELOG.md:161]: I'm not sure this is an entry that a typical user would understand. `kongstate.Override` is an implementation detail isn't it?

What is the user facing change here? Can we rephrase it to have a user facing meaning?

Comment (randmonkey) [reply CHANGELOG.md:161]: User will see the `protocols` moved from expressions to `protocols` field of translated Kong route.

Comment (randmonkey) [reply CHANGELOG.md]: updated.

**Review by pmalek — APPROVED** [review]
🚢

**Review by czeslavo — APPROVED** [review]
